### ISSUE TITLE
freerdp-shadow-cli disconnects immediately 

### ISSUE THREAD

**Issue by stdedos** [FreeRDP/FreeRDP#6231]
**Describe the bug**
Using `freerdp-shadow-cli` with non-NLA auth disconnects immediately, with or without force-providing password. No RDP-initiated password prompt.

```console
$ DISPLAY=:0 freerdp-shadow-cli /sec:rdp
[12:03:06:649] [21920:21920] [INFO][com.freerdp.server.shadow.x11] - X11 Extensions: XFixes: 1 Xinerama: 1 XDamage: 0 XShm: 0
[12:03:06:660] [21920:21920] [INFO][com.freerdp.core.listener] - Listening on 0.0.0.0:3389
[12:03:10:530] [21920:21975] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[12:03:10:530] [21920:21975] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:0 RDP:1
[12:03:10:530] [21920:21975] [ERROR][com.freerdp.core.connection] - server supports only Standard RDP Security
[12:03:10:530] [21920:21975] [ERROR][com.freerdp.core.connection] - Protocol security negotiation failure
[12:03:10:530] [21920:21975] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_INITIAL - rdp_server_accept_nego() fail
[12:03:10:530] [21920:21975] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[12:03:10:530] [21920:21975] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
[12:03:12:276] [21920:21998] [INFO][com.freerdp.core.connection] - Client Security: NLA:0 TLS:0 RDP:1
[12:03:12:276] [21920:21998] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:0 RDP:1
[12:03:12:276] [21920:21998] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:0 RDP:1
[12:03:16:646] [21920:21998] [INFO][com.freerdp.core.connection] - Accepted client: 
[12:03:16:646] [21920:21998] [INFO][com.freerdp.core.connection] - Accepted channels:
[12:03:16:646] [21920:21998] [INFO][com.freerdp.core.connection] -  rdpdr
[12:03:16:646] [21920:21998] [INFO][com.freerdp.core.connection] -  rdpsnd
[12:03:16:646] [21920:21998] [INFO][com.freerdp.core.connection] -  cliprdr
[12:03:16:646] [21920:21998] [INFO][com.freerdp.core.connection] -  drdynvc
[12:03:16:646] [21920:21998] [INFO][com.freerdp.core.gcc] - Active rdp encryption level: CLIENT-COMPATIBLE
[12:03:16:646] [21920:21998] [INFO][com.freerdp.core.gcc] - Selected rdp encryption method: 128BIT
[12:03:16:883] [21920:21998] [INFO][com.freerdp.client.shadow] - Client from  is activated (2560x1440@32)
[12:03:16:883] [21920:21998] [ERROR][com.freerdp.client.shadow] - client authentication failure: -1
[12:03:16:883] [21920:21998] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_ACTIVE - peer_recv_pdu() fail
[12:03:16:883] [21920:21998] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[12:03:16:883] [21920:21998] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor

```

`-auth` partially works (only shows 1 monitor), but I guess it's not a workaround

**Application details**
* FreeRDP version (`xfreerdp /version`): 2.0.0-dev5
* OS version connecting to (server side): ubuntu xenial
* If available the log output from a run with `/log-level:trace 2>&1 | tee log.txt` 

**Environment (please complete the following information):**
 - OS: ubuntu xenial
 - Architecture: [amd64, arm]:amd64

_Note: Some people get confused by the options - they think that `/sec-tls` is supposed to work (or used to work at some point, idk)_

**Comment by akallabeth** [comment]
@stdedos
1. check if you compiled with `PAM` enabled, aka look for a line in the cmake configuration step or check `/buildconfig`
1. Check the server side log (`export WLOG_LEVEL=debug`) to see what exactly is failing.
1. You might also try with a newer version, we´ve fixed a lot since `xenial` (I assume this is the default package from the repos?)

**Comment by stdedos** [comment]
> @stdedos
> 
> 1. check if you compiled with `PAM` enabled, aka look for a line in the cmake configuration step or check `/buildconfig`
> 3. You might also try with a newer version, we´ve fixed a lot since `xenial` (I assume this is the default package from the repos?)

I haven't compiled anything (https://github.com/FreeRDP/FreeRDP/issues/6230), so I wouldn't know. I would assume the packages would come with "such a cornerstone library" as PAM.

_Verified `WITH_PAM=ON`_

> 2. Check the server side log (`export WLOG_LEVEL=debug`) to see what exactly is failing.


```console
u@h:~$ WLOG_LEVEL=debug freerdp-shadow-cli
[11:21:14:193] [27255:27255] [INFO][com.freerdp.server.shadow.x11] - X11 Extensions: XFixes: 1 Xinerama: 1 XDamage: 0 XShm: 0
[11:21:14:211] [27255:27255] [INFO][com.freerdp.core.listener] - Listening on 0.0.0.0:3389
[11:21:36:735] [27255:27256] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[11:21:36:737] [27255:27546] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x0000000B
[11:21:36:737] [27255:27546] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[11:21:36:737] [27255:27546] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:1 RDP:1
[11:21:36:737] [27255:27546] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:1 RDP:0
[11:21:36:392] [27255:27546] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
[11:21:36:392] [27255:27546] [ERROR][com.freerdp.core] - transport_read_layer:freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[11:21:36:392] [27255:27546] [DEBUG][com.freerdp.core.transport] - transport_check_fds: transport_read_pdu() - -1
[11:21:36:392] [27255:27546] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[11:21:36:392] [27255:27546] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
[11:21:36:492] [27255:27256] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[11:21:36:498] [27255:27558] [DEBUG][com.freerdp.core.nego] - received cookie [Cookie: mstshash=usr]
[11:21:36:498] [27255:27558] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x0000000B
[11:21:36:498] [27255:27558] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[11:21:36:498] [27255:27558] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:1 RDP:1
[11:21:36:498] [27255:27558] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:1 RDP:0
[11:21:37:877] [27255:27558] [INFO][com.freerdp.core.connection] - Accepted client: CLI
[11:21:37:877] [27255:27558] [INFO][com.freerdp.core.connection] - Accepted channels:
[11:21:37:877] [27255:27558] [INFO][com.freerdp.core.connection] -  rdpdr
[11:21:37:877] [27255:27558] [INFO][com.freerdp.core.connection] -  rdpsnd
[11:21:37:877] [27255:27558] [INFO][com.freerdp.core.connection] -  cliprdr
[11:21:37:877] [27255:27558] [INFO][com.freerdp.core.connection] -  drdynvc
[11:21:37:877] [27255:27558] [INFO][com.freerdp.core.gcc] - Active rdp encryption level: NONE
[11:21:37:877] [27255:27558] [INFO][com.freerdp.core.gcc] - Selected rdp encryption method: NONE
[11:21:37:592] [27255:27558] [DEBUG][com.freerdp.core.info] - rdp client address: [IP]
[11:21:37:592] [27255:27558] [DEBUG][com.freerdp.core.info] - rdp client dir: [C:\Windows\system32\mstscax.dll]
[11:21:37:641] [27255:27558] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x1f size=37 channelId=1009)
[11:21:37:642] [27255:27558] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[11:21:37:642] [27255:27558] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[11:21:37:642] [27255:27558] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x28 size=41 channelId=1009)
[11:21:37:642] [27255:27558] [INFO][com.freerdp.client.shadow] - Client from IP is activated (4160x1440@32)
[11:21:37:642] [27255:27558] [ERROR][com.freerdp.client.shadow] - client authentication failure: -1
[11:21:37:642] [27255:27558] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_ACTIVE - peer_recv_pdu() fail
[11:21:37:642] [27255:27558] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[11:21:37:642] [27255:27558] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[11:21:37:642] [27255:27558] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
^C
u@h:~$ export WLOG_LEVEL=debug
u@h:~$ freerdp-shadow-cli
[11:21:59:009] [28034:28034] [INFO][com.freerdp.server.shadow.x11] - X11 Extensions: XFixes: 1 Xinerama: 1 XDamage: 0 XShm: 0
[11:21:59:023] [28034:28034] [INFO][com.freerdp.core.listener] - Listening on 0.0.0.0:3389
[11:22:01:567] [28034:28035] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[11:22:01:577] [28034:28075] [DEBUG][com.freerdp.core.nego] - received cookie [Cookie: mstshash=usr]
[11:22:01:577] [28034:28075] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x0000000B
[11:22:01:577] [28034:28075] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[11:22:01:577] [28034:28075] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:1 RDP:1
[11:22:01:577] [28034:28075] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:1 RDP:0
[11:22:01:662] [28034:28075] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
[11:22:01:662] [28034:28075] [ERROR][com.freerdp.core] - transport_read_layer:freerdp_set_last_error_ex ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[11:22:01:662] [28034:28075] [DEBUG][com.freerdp.core.transport] - transport_check_fds: transport_read_pdu() - -1
[11:22:01:662] [28034:28075] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[11:22:01:662] [28034:28075] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
[11:22:02:689] [28034:28035] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[11:22:02:691] [28034:28076] [DEBUG][com.freerdp.core.nego] - received cookie [Cookie: mstshash=usr]
[11:22:02:691] [28034:28076] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x0000000B
[11:22:02:691] [28034:28076] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[11:22:02:691] [28034:28076] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:1 RDP:1
[11:22:02:691] [28034:28076] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:1 RDP:0
[11:22:02:805] [28034:28076] [INFO][com.freerdp.core.connection] - Accepted client: CLI
[11:22:02:805] [28034:28076] [INFO][com.freerdp.core.connection] - Accepted channels:
[11:22:02:805] [28034:28076] [INFO][com.freerdp.core.connection] -  rdpdr
[11:22:02:805] [28034:28076] [INFO][com.freerdp.core.connection] -  rdpsnd
[11:22:02:805] [28034:28076] [INFO][com.freerdp.core.connection] -  cliprdr
[11:22:02:805] [28034:28076] [INFO][com.freerdp.core.connection] -  drdynvc
[11:22:02:805] [28034:28076] [INFO][com.freerdp.core.gcc] - Active rdp encryption level: NONE
[11:22:02:805] [28034:28076] [INFO][com.freerdp.core.gcc] - Selected rdp encryption method: NONE
[11:22:02:338] [28034:28076] [DEBUG][com.freerdp.core.info] - rdp client address: [IP]
[11:22:02:338] [28034:28076] [DEBUG][com.freerdp.core.info] - rdp client dir: [C:\Windows\system32\mstscax.dll]
[11:22:02:416] [28034:28076] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x1f size=37 channelId=1009)
[11:22:02:416] [28034:28076] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[11:22:02:417] [28034:28076] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=41 channelId=1009)
[11:22:02:422] [28034:28076] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x28 size=41 channelId=1009)
[11:22:02:422] [28034:28076] [INFO][com.freerdp.client.shadow] - Client from IP is activated (4160x1440@32)
[11:22:02:422] [28034:28076] [ERROR][com.freerdp.client.shadow] - client authentication failure: -1
[11:22:02:422] [28034:28076] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_ACTIVE - peer_recv_pdu() fail
[11:22:02:422] [28034:28076] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[11:22:02:422] [28034:28076] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[11:22:02:422] [28034:28076] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
^C
```

**Comment by akallabeth** [comment]
This looks like everything is working but `x11_shadow_pam_authenticate` can not authenticate the user (or is missing some environment?)
Maybe `x11_shadow_pam_get_service_name` is the culprit?

**Comment by stdedos** [comment]
That could very well be possible, given that I am running this remotely (no point in being in front of your screen and start a shadow service, right?).

Also, I have tried running this dead from `DISPLAY:=0` and it gave me a similar-looking error (I thought I had reported both of these issues last time I was in IRC and received no replies, but idk where everything went)

**Comment by akallabeth** [comment]
@stdedos added some error logging and better error checks with #6239 might give you a better hint what is missing.
As this is not identified as authentication (configuration) problem, I´ll close this.

**Comment by stdedos** [comment]
Fingers crossed that repositories are fixed too. At least previous major/minor and next would be nice to be pre-compiled.

**Comment by stdedos** [comment]
No extra diagnostics are coming from today's nightly build:

```
Index of /repositories/deb/xenial/pool/main/f/freerdp-nightly/
../
freerdp-nightly-dbg_2.0.0+0~20200606014824.824~..> 06-Jun-2020 02:39             8936606
freerdp-nightly-dbg_2.0.0+0~20200606014824.824~..> 06-Jun-2020 02:09             8579750
freerdp-nightly-dev_2.0.0+0~20200606014824.824~..> 06-Jun-2020 02:39              233576
freerdp-nightly-dev_2.0.0+0~20200606014824.824~..> 06-Jun-2020 02:09              233632
freerdp-nightly_2.0.0+0~20200606014824.824~1.gb..> 06-Jun-2020 02:39                1565
freerdp-nightly_2.0.0+0~20200606014824.824~1.gb..> 06-Jun-2020 02:39             4492016
freerdp-nightly_2.0.0+0~20200606014824.824~1.gb..> 06-Jun-2020 02:39             2256574
freerdp-nightly_2.0.0+0~20200606014824.824~1.gb..> 06-Jun-2020 02:09             2157902
```

Please re-open

**Comment by stdedos** [comment]
_ping @akallabeth_

**Comment by stdedos** [comment]
```console
$ WLOG_LEVEL=debug DISPLAY=:0 freerdp-shadow-cli /sec:rdp
[11:08:53:251] [2686:2686] [INFO][com.freerdp.server.shadow.x11] - X11 Extensions: XFixes: 1 Xinerama: 1 XDamage: 0 XShm: 0
[11:08:53:261] [2686:2686] [INFO][com.freerdp.core.listener] - Listening on [0.0.0.0]:3389
[11:09:01:840] [2686:2691] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[11:09:01:842] [2686:3028] [DEBUG][com.freerdp.core.nego] - received cookie [Cookie: mstshash=usr]
[11:09:01:842] [2686:3028] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x0000000B
[11:09:01:842] [2686:3028] [INFO][com.freerdp.core.connection] - Client Security: NLA:1 TLS:1 RDP:0
[11:09:01:842] [2686:3028] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:0 RDP:1
[11:09:01:842] [2686:3028] [ERROR][com.freerdp.core.connection] - server supports only Standard RDP Security
[11:09:01:842] [2686:3028] [ERROR][com.freerdp.core.connection] - Protocol security negotiation failure
[11:09:01:842] [2686:3028] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_INITIAL - rdp_server_accept_nego() fail
[11:09:01:842] [2686:3028] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[11:09:01:842] [2686:3028] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[11:09:01:842] [2686:3028] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
[11:09:02:237] [2686:2691] [DEBUG][com.winpr.thread] - Thread running, setting to detached state!
[11:09:02:242] [2686:3077] [DEBUG][com.freerdp.core.nego] - received cookie [Cookie: mstshash=usr]
[11:09:02:242] [2686:3077] [DEBUG][com.freerdp.core.nego] - RDP_NEG_REQ: RequestedProtocol: 0x00000000
[11:09:02:242] [2686:3077] [INFO][com.freerdp.core.connection] - Client Security: NLA:0 TLS:0 RDP:1
[11:09:02:242] [2686:3077] [INFO][com.freerdp.core.connection] - Server Security: NLA:0 TLS:0 RDP:1
[11:09:02:242] [2686:3077] [INFO][com.freerdp.core.connection] - Negotiated Security: NLA:0 TLS:0 RDP:1
[11:09:02:278] [2686:3077] [INFO][com.freerdp.core.connection] - Accepted client: CLI
[11:09:02:278] [2686:3077] [INFO][com.freerdp.core.connection] - Accepted channels:
[11:09:02:278] [2686:3077] [INFO][com.freerdp.core.connection] -  rdpdr
[11:09:02:278] [2686:3077] [INFO][com.freerdp.core.connection] -  rdpsnd
[11:09:02:278] [2686:3077] [INFO][com.freerdp.core.connection] -  cliprdr
[11:09:02:278] [2686:3077] [INFO][com.freerdp.core.connection] -  drdynvc
[11:09:02:278] [2686:3077] [INFO][com.freerdp.core.gcc] - Active rdp encryption level: CLIENT-COMPATIBLE
[11:09:02:278] [2686:3077] [INFO][com.freerdp.core.gcc] - Selected rdp encryption method: 128BIT
[11:09:03:871] [2686:3077] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x1f size=49 channelId=1009)
[11:09:03:872] [2686:3077] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=53 channelId=1009)
[11:09:03:872] [2686:3077] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x14 size=53 channelId=1009)
[11:09:03:874] [2686:3077] [DEBUG][com.freerdp.core.rdp] - rdp_send_data_pdu: sending data (type=0x28 size=53 channelId=1009)
[11:09:03:874] [2686:3077] [INFO][com.freerdp.client.shadow] - Client from IP is activated (2560x1440@32)
[11:09:03:874] [2686:3077] [ERROR][com.freerdp.client.shadow] - client authentication failure: -1
[11:09:03:874] [2686:3077] [ERROR][com.freerdp.core.peer] - peer_recv_callback: CONNECTION_STATE_ACTIVE - peer_recv_pdu() fail
[11:09:03:874] [2686:3077] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[11:09:03:875] [2686:3077] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[11:09:03:875] [2686:3077] [ERROR][com.freerdp.client.shadow] - Failed to check FreeRDP file descriptor
```

**Comment by akallabeth** [comment]
Bug in nightly that builds the package without PAM support, should be fixed with #6325
